marlowe/src/Language/Marlowe/Client.hs

```diff
@@ -240,13 +240,18 @@ marlowePlutusContract = selectList [create, apply, auto, redeem, close]
     create = endpoint @"create" $ \(reqId, owners, contract) -> catchError reqId "create" $ do
         -- Create a transaction with the role tokens and pay them to the contract creator
         -- See Note [The contract is not ready]
+        ownPubKey <- Contract.ownPubKeyHash
         (params, distributeRoleTokens, lkps) <- setupMarloweParams owners contract
         slot <- currentSlot
         let StateMachineClient{scInstance} = mkMarloweClient params
         let marloweData = MarloweData {
                 marloweContract = contract,
-                marloweState = emptyState slot }
-        let payValue = adaValueOf 0
+                marloweState = State
+                    { accounts = AssocMap.singleton (PK ownPubKey, Token adaSymbol adaToken) 1000000
```

Comment:
If I understand this correctly, now each role token can have some balance and the contract creator will have the min ada assigned to its public key hash instead of his role, right? This would allow a user to create a contract even if it doesn't have a role assigned. Neat 👍

Reply:
Could you add a comment and change `1000000` with a more descriptive constant, or even better, a parameter?

Reply:
While working on [SCP-3099 branch](https://github.com/input-output-hk/marlowe-cardano/tree/SCP-3099) to upgrade `main` to use the latest `plutus-apps`, I encounted test failures for the zero-coupon bond because of the min-ADA requirement. These failures also occur on `anemish/one-ada-limit` with the latest `plutus-apps`, which now correctly enforces the min-ADA requirement.

This isn't well publicized, but min-ADA is greater that 1,000,000 lovelace *if there is a datum hash in the UTxO*, even if there are no native tokens in the UTxO.

    # Min-ADA w/o a datum.
    
    $ cardano-cli transaction calculate-min-required-utxo  \
      --alonzo-era --protocol-params-file testnet.protocol \
      --tx-out addr_test1vq9prvx8ufwutkwxx9cmmuuajaqmjqwujqlp9d8pvg6gupczgtm9j+10000000
    
    Lovelace 999978
    
    
    # Min-ADA w/ a datum.
    
    $ cardano-cli transaction calculate-min-required-utxo  \
      --alonzo-era --protocol-params-file testnet.protocol \
      --tx-out-datum-hash-file example.datum               \
      --tx-out addr_test1vq9prvx8ufwutkwxx9cmmuuajaqmjqwujqlp9d8pvg6gupczgtm9j+10000000
    
    Lovelace 1344798
    
    
    # Node and CLI version.
    
    $ cardano-cli --version
    cardano-cli 1.31.0 - linux-x86_64 - ghc-8.10
    git rev 2cbe363874d0261bc62f52185cf23ed492cf4859

The safest way to compute this is with `Cardano.Api.calculateMinimumUTxO`.

Reply:
Here is a little more information about min-ADA:
1. [`Plutus.Ledger.Index.minAdaTxOut`](https://github.com/input-output-hk/plutus-apps/blob/a6868cde87600051a4cc808777d665d0b859853c/plutus-ledger/src/Ledger/Index.hs#L378-L379) uses 2 ADA as a generic min-ADA.
2. Given the current protocol parameters, [min-ADA](https://mantis.functionally.io/how-to/min-ada-value/) could be as high as 22.5 ADA. This worst-case scenario corresponds to the situation where there are 123 native tokens with 32-byte names on deposit in the Marlowe script.